src/border_agent/border_agent.cpp

```diff
@@ -71,6 +71,9 @@ namespace otbr {
 
 static const char    kBorderAgentServiceType[]    = "_meshcop._udp"; ///< Border agent service type of mDNS
 static constexpr int kBorderAgentServiceDummyPort = 49152;
+#if OTBR_ENABLE_EPSKC
```

Comment:
Would you mind omitting this guard? We'd better to control the on/off at runtime. We can keep it as a the default value of whether the feature is on or off.

Reply:
Per discussion, will enable feature flag to control ePSKc (tracking b/339768305);

Reply:
Let's not guard the feature compiling with this macro, unless you want to add an additional test verifying that when `OTBR_ENABLE_EPSKC` is 0, at the source can still be compiled.

Reply:
Modified and accepted. Thanks~

Reply:
A reminder here to obsolete OTBR_EPSKC option along with introducing runtime on/off control. @mia1yang